plugins/package-managers/pub/src/main/kotlin/Pubspec.kt

```diff
@@ -77,7 +77,7 @@ internal data class Pubspec(
     @Serializable
     data class HostedDependency(
         val version: String,
-        val url: String? = null
+        val hosted: String? = null
```

Comment:
Hosted packages can be represented in multiple ways. 
The link above only points to one of the representations. 
There is a way to specify a `url` for hosted dependencies. 
So, the data class should IMO remain as-is.

Reply:
copied from the docs:
```
environment:
  sdk: '>=2.14.0 < 3.0.0'

dependencies:
  transmogrify:
    hosted:
      name: transmogrify
      url: https://some-package-server.com
    version: ^1.4.0
```

Reply:
The link you provided points to the short form, which uses a primitive type to specify the `url` in a more compact way.

Reply:
Hmm. The [link](https://dart.dev/tools/pub/dependencies#hosted-packages) IMO docuemnts three forms of hosted packages:

1. With a scalar value:

```
dependencies:
  transmogrify: ^1.4.0
```

2. With a map value that contains `hosted` and `version` *on the same level*, while `hosted` having a scalar value:

```
dependencies:
  transmogrify:
    hosted: https://some-package-server.com
    version: ^1.4.0
```

3. With a map value where also `hosted` is a map (only relevant for an SDK version earlier than 2.15):

```
dependencies:
  transmogrify:
    hosted:
      name: transmogrify
      url: https://some-package-server.com
    version: ^1.4.0
```

My goal was to support case 2 where `hosted` and `version` are siblings as I did not see this case being covered. Am I missing something?

Reply:
To add to that, the naming in this case actually does not matter too much as these classes are not directly used via their auto-generated serializers, but the `DependencyMapSerializer` constructs these classes manually, and takes care about the 3. case by explicitly looking for `hosted.get<YamlScalar>("url")`.

Anyway, if the classes *were* deserialized directly, I believe naming the property `hosted` instead of `url` to be more correct.

Reply:
I would prefer to keep `url`, because
1. In the most detailed representation the name is `url`
2. Hosted is name of the class. So it's kind of `Hosted.hosted` which seems a tad odd
3. Considering the values, `url` seems to fit better.
4. It's debatable whether the version values are actually part of the `hosted` object. It seems to rather be a sibling.

Reply:
I still don't believe it makes sense to stick with `url` in the longer term as it would break as soon as we'd be able to deserialize classes directly (see https://github.com/charleskorn/kaml/issues/29 and / or https://github.com/charleskorn/kaml/issues/612), but for the sake of moving things forward I'll drop that commit.